identity-account/src/stronghold/context.rs

```diff
@@ -415,62 +403,62 @@ impl Runtime {
     }
   }
 
-  fn password(&self, path: &Path) -> Result<Password> {
+  fn password(&self, path: &Path) -> IotaStrongholdResult<Password> {
     self
       .password_store()?
       .get(path)
       .map(|(password, _)| *password)
-      .ok_or(Error::StrongholdPasswordNotSet)
+      .ok_or(StrongholdError::StrongholdPasswordNotSet)
   }
 
-  fn password_elapsed(&self, path: &Path) -> Result<Option<Duration>> {
+  fn password_elapsed(&self, path: &Path) -> IotaStrongholdResult<Option<Duration>> {
     self
       .password_store()
       .map(|store| store.get(path).map(|(_, interval)| interval.elapsed()))
   }
 
-  fn set_password(&self, path: &Path, password: Password) -> Result<()> {
+  fn set_password(&self, path: &Path, password: Password) -> IotaStrongholdResult<()> {
     self
       .password_store()?
       .insert(path.to_path_buf(), (password, Instant::now()));
 
     Ok(())
   }
 
-  fn set_password_access(&self, path: &Path) -> Result<()> {
+  fn set_password_access(&self, path: &Path) -> IotaStrongholdResult<()> {
     if let Some((_, ref mut time)) = self.password_store()?.get_mut(path) {
       *time = Instant::now();
     } else {
-      return Err(Error::StrongholdPasswordNotSet);
+      return Err(StrongholdError::StrongholdPasswordNotSet);
     }
 
     Ok(())
   }
 
-  fn set_password_clear(&self, interval: Duration) -> Result<()> {
+  fn set_password_clear(&self, interval: Duration) -> IotaStrongholdResult<()> {
     *self.password_clear()? = interval;
 
     Ok(())
   }
 
-  fn event_listeners(&self) -> Result<MutexGuard<'_, Vec<Listener>>> {
+  fn event_listeners(&self) -> IotaStrongholdResult<MutexGuard<'_, Vec<Listener>>> {
     self
       .event_listeners
       .lock()
-      .map_err(|_| Error::StrongholdMutexPoisoned("listeners"))
+      .map_err(|_| StrongholdError::StrongholdMutexPoisoned("listeners"))
   }
```

Comment:
Do we want to give the guard and allow access if the mutex is poisoned? it sounds to me more of an error case, not sure if we want to ignore it completely.

Reply:
I'm not entirely sure what to do here. A typical solution is to unwrap on the `.lock()` call, so if some other thread panicked, we panic as well. The current solution returns an error, and that at least gives the caller a chance to react to it, although it seems unlikely to me that anyone would do so in practice. Basically, if some other part of the program already panicked, which we try to avoid in the first place, then it seems okay-ish to call `unwrap()`, because the error in the other thread was already severe enough to warrant a panic.
Using `PoisonError::into_inner` grants access to the `MutexGuard` despite the lock being poisoned, but it's a bit unclear to me whether the lock semantics are still being upheld, e.g. can two threads that haven't panicked access the data concurrently? I strongly assume not, in which case doing this seems okay.

I'm mainly trying to get rid of the error variant, because it doesn't neatly fall into a category of errors, which is what we will have to come up with to implement a more generic/abstract `StorageError` for the storage interface, but that's probably not that important to consider here.

Does anyone else have an opinion on what's best here, @cycraig or @olivereanderson?

Reply:
@olivereanderson had a neat suggestion, which is to use the `parking_lot::Mutex` instead, which does not have poisoning in the first place, so we can avoid the error without a potential panic. `parking_lot` is not a direct dependency of the account yet, but it is used by our transitive dependencies (like `tokio`) already, so adding it seems totally fine.

Reply:
Thanks Philipp and Oliver for the suggestion. I replaced it with `parking_lot::Mutex`.